### PR TITLE
fix shadowed variables

### PR DESCRIPTION
**Pull request by lazydroid**
this MR is a preparation to use `-Werror=shadow` on the whole project for the purpose to catching shadowed variables using compiler instead of people. currently all variables that shadow existing class members are renamed into "var_" (notice the underscore) -- this usually happens in the constructors.

in all other cases, the variables are renamed or removed or the code is refactored to avoid using the same names several times in the same code block (C++). I have tried to keep the code intact as much as possible, so there are only minor changes, but there are plenty, sorry about that.

I have made sure the pybind11 bindings are working and Taiju helped with boost::python bindings, hopefully everything is well and nothing is broken. The image is built, and once the tests pass, I'll push this MR for the reviews.

This PR has started from "master" branch, then "production" branch was merged in. Now we're targeting "production", not "master".

### CODE REVIEW

**Comment by cielavenir** [comment]
changed the base branch to production.

**Comment by cielavenir** [comment]
btw, it should be time to have production merged into master again.

File: plugins/rplanners/parabolicsmoother2.cpp [hunk 1]
```diff
@@ -2999,7 +2999,7 @@ class ParabolicSmoother2 : public PlannerBase, public RampOptimizer::Feasibility
 #endif
                                         for( size_t j = 0; j < vellimits.size(); ++j ) {
                                             dReal fMinVelLimit = max(RaveFabs(v0Vect[j]), RaveFabs(v1Vect[j]));
-                                            dReal fVelMult = RaveSqrt(retcheck.vReductionFactors[j]);
+                                            dReal fVelMult1 = RaveSqrt(retcheck.vReductionFactors[j]);
                                             if( vellimits[j] * fVelMult < fMinVelLimit ) {
```
[the remark this answers is not in the document]
Comment (lazydroid) [reply plugins/rplanners/parabolicsmoother2.cpp:3003]: fixed, thanks!

**Comment by lazydroid** [comment]
> changed the base branch to production.

now it's a bit more manageable, thanks! =)

File: plugins/rplanners/parabolicsmoother2.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (Puttichai) [reply plugins/rplanners/parabolicsmoother2.cpp:3003]: I think you should just remove the declaration here and make it use the already declared `fVelMult`. The following if statement should be using this multiplier.

File: src/libopenrave/configurationspecification.cpp [hunk 2]
```diff
@@ -412,7 +412,7 @@ void ConfigurationSpecification::AddDerivativeGroups(int deriv, bool adddeltatim
     bool hasdeltatime = false;
     FOREACH(itgroup,_vgroups) {
         string replacename;
-        int offset = -1;
+        offset = -1; // we have just set `offset` to `GetDOF()`, shall it be reset to -1?
```
Comment (cielavenir) [inline src/libopenrave/configurationspecification.cpp:415]: offset is used below, so please still declare the variable and move `int offset = GetDOF();` below FOREACH.

Comment (cielavenir) [reply src/libopenrave/configurationspecification.cpp]: done here

File: plugins/rplanners/parabolicsmoother2.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (cielavenir) [reply plugins/rplanners/parabolicsmoother2.cpp]: (then I wonder why fVelMult was shadowed here)

**Comment by lazydroid** [comment]
@cielavenir thanks! should be good to merge!

File: plugins/rplanners/parabolicsmoother2.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (lazydroid) [reply plugins/rplanners/parabolicsmoother2.cpp:3003]: > I think you should just remove the declaration here and make it use the already declared `fVelMult`. The following if statement should be using this multiplier.

removed the declaration, pipeline #317310 passed. please, review again.

**Comment by cielavenir** [comment]
if possible please check https://github.com/rdiankov/openrave/pull/1070 quickly

**Comment by rdiankov** [comment]
thanks

**Pull request merged**